### DISCUSSION TITLE
Branches and tags with the same name

### DISCUSSION THREAD

**Discussion by cpretzer** [argoproj/argo-cd#12461] [General]
What is the expected behavior for ArgoCD when the `targetRevision` is set to a tag that has the same name as a branch?

For example, let's say there are a tag and a branch named both with the name `release-0.1.0` and the `targetRevision` of the Application is set to `release-0.1.0`. The behavior that I saw was that the tag took precedence and the branch was completely ignored.

A simple workaround is to make sure that the tags are named differently from branches, but I'm curious to know if this is the intended behavior.

**Comment by jannfis** [comment]
Hm, that's a good question. From looking at the code, if I understanding it correctly, we depend on the semantics of go-git because we use that to resolve references in Git and use whichever "comes first".

The official Git tool behaves different, though: If there's a tag and a branch with the same name, it will use the branch and emit a warning about using an ambiguous reference.

Have you tried to specify `refs/heads/release-0.1.0` to track the revision? This should point to the head of that branch, and is non-ambiguous.

**Comment by cpretzer** [comment]
Thanks @jannfis I was thinking of trying to use the fully qualified ref. I'll give it a shot and report back.
